src/main/java/net/snowflake/ingest/streaming/internal/ParquetFlusher.java

```diff
@@ -116,7 +120,14 @@ private SerializationResult serializeFromJavaObjects(
     // We insert the filename in the file itself as metadata so that streams can work on replicated
     // mixed tables. For a more detailed discussion on the topic see SNOW-561447 and
     // http://go/streams-on-replicated-mixed-tables
-    metadata.put(Constants.PRIMARY_FILE_ID_KEY, StreamingIngestUtils.getShortname(filePath));
+    // Using chunk offset as suffix ensures that for interleaved tables, the file
+    // id key is unique for each chunk. Each chunk is logically a separate Parquet file that happens
+    // to be bundled together.
+    String shortName = StreamingIngestUtils.getShortname(filePath);
+    final String[] parts = shortName.split("\\.");
+    metadata.put(
+        Constants.PRIMARY_FILE_ID_KEY,
+        String.format("%s%s.%s", parts[0], chunkStartOffset, parts[1]));
```

Comment:
Is there any issue if the does not match with the file name? Is there any server side change this needs to wait before merging? cc: @sfc-gh-kkloudas

Reply:
And do you think we can do %s_%s.%s?

Reply:
I am trying to match with what Andrey will do on the server side - we don't have to match but would be good to be consistent. Let's have @sfc-gh-azagrebin confirm what he'd do. We can wait for the server side change. And no, it does not need to match the file name at all - it can be a UUID for all change tracking cares about.

Reply:
And if we change the format, is there a use case that customer needs to join with previous values as part of MERGE?

Reply:
No, this value is supposed to be unique that does not appear in any other files.

Reply:
I would also vote for this:
```
And do you think we can do %s_%s.%s?
```
let me also first try the server side change

Reply:
just in case to reduce any potential blast radius, I would append only non-zero offset.

Reply:
We just discussed it with @sfc-gh-lpapke. We have to add "_" because the name w/o extension ends with a number in our case and we can get into a similar problem that was mentioned for regular partitions:
name_10.bdec  + offset 110 = name_10110.bdec
name_101.bdec  + offset 10 = name_10110.bdec
those 2 are different bdecs but they will be the same PPNs.

Reply:
Good catch. I added an `_` separator before the `.bdec` suffix. PTAL. I have also added a check for 0 chunk offset.